gnosis/cowswap
# [1689 - Quote endpoint] New Quote endpoint for price & fee

Closes #1689 

Fetches both FEE and PRICE in one endpoint, relies on different parameters.

### TESTING
_Right now, the new quote endpoint for getting PRICE is disabled, legacy price racing is being used_
1. Open the Networks tab in your browser's dev tools
2. filter by `quote`
3. select any token pair and set an amount
4. look at the `quote` post requests made to the GP backend
5. click on one and stay inside the `headers` tab
6. scroll to the bottom and check the body object being `POST`ed
7. confirm that the parameters are all filled in
8. make sure that the `validTo` param (denoted in SECONDS) converted into milliseconds (rounded down) is a valid date in the console that can be done via `Date(Math.floor(<validToParam> * 1000))` e.g `validTo` returns `1637059250` - so type in `Date(Math.floor(1637059250 * 1000))` and make sure you get a string date like: `'Tue Nov 16 2021 10:41:25 GMT+0000 (Western European Standard Time)'` and that it is in the future
9. price and quote should be working with no new issues differing from previous versions
10. errors that occur should be properly handled as before

## TESTING (LATEST)
1. you can manually change the strategy type in the console window
2. chrome console
3. `setStrategy('LEGACY')` or `setStrategy('COWSWAP')`
